core/src/main/java/org/apache/brooklyn/core/location/access/PortForwardManagerImpl.java

```diff
@@ -83,9 +83,6 @@
 
     private final Map<AssociationListener, Predicate<? super AssociationMetadata>> associationListeners = new ConcurrentHashMap<AssociationListener, Predicate<? super AssociationMetadata>>();
 
-    @Deprecated
-    protected final Map<String,String> publicIpIdToHostname = new LinkedHashMap<String,String>();
```

Comment:
@sjcorbett my strategy when deprecating fields is to mark it deprecated and transient for one major release, make sure all accessors transform it, and make sure at least one accessor gets exercised.

this should ensure that if state from one version is carried through subsequent major versions it will migrate.

however there is not currently any way to force exercising in all cases i don't think (we could introduce a method `postRebind()` for that purpose?).  for now i suggest keep it in perpetuity with a comment `// kept for persistence from previous versions; see https://github.com/apache/brooklyn-server/pull/318`